modules/PathUtils.js

```diff
@@ -4,8 +4,11 @@ export const addLeadingSlash = (path) =>
 export const stripLeadingSlash = (path) =>
   path.charAt(0) === '/' ? path.substr(1) : path
 
-export const stripPrefix = (path, prefix) =>
-  path.indexOf(prefix) === 0 ? path.substr(prefix.length) : path
+export const hasBasename = (path, prefix) => 
+  (new RegExp('^' + prefix + '(\\/|\\?|#|$)', 'i')).test(path)
```

Comment:
This doesn't work if `prefix` has regex special symbols in them. Like `$`. It should be
```
export const hasBasename = (path, prefix) => path.startsWith(prefix)
```

Reply:
A `basename` is meant to represent full segments, which `startsWith` cannot enforce on its own (the basename `/test` should not match the pathname `/testing`).

The `prefix` probably should be passed to a function that escapes any special regexp characters. I'm sure that a PR with tests would be appreciated for this.

Reply:
Thanks for the quick reply. I realised my mistake while makinging the fix. I created this PR for escaping special regex symbols: https://github.com/ReactTraining/history/pull/544